Can't install tensorflow because of bazel

I am trying to install tensor flow for cpu via spack.  It falls over at bazel:
```
==> Installing bazel
==> Searching for binary cache of bazel
==> No binary for bazel found: installing from source
==> Using cached archive: /lus/theta-fs0/projects/datascience/cadams/spack/cooley/spack/var/spack/cache/_source-cache/archive/c0/c0e94f8f818759f3f67af798c38683520c540f469cb41aea8f5e5a0e43f11600.zip
==> Staging archive: /tmp/cadams/spack-stage/spack-stage-bazel-0.26.1-4mmpkcidyja4i6smxm2ahqifvh322xsz/bazel-0.26.1-dist.zip
==> Created stage in /tmp/cadams/spack-stage/spack-stage-bazel-0.26.1-4mmpkcidyja4i6smxm2ahqifvh322xsz
==> Applied patch /lus/theta-fs0/projects/datascience/cadams/spack/cooley/spack/var/spack/repos/builtin/packages/bazel/bazelruleclassprovider-0.25.patch
==> Applied patch /lus/theta-fs0/projects/datascience/cadams/spack/cooley/spack/var/spack/repos/builtin/packages/bazel/unix_cc_configure-0.15.patch
==> Applied patch /lus/theta-fs0/projects/datascience/cadams/spack/cooley/spack/var/spack/repos/builtin/packages/bazel/compile-0.21.patch
==> Building bazel [Package]
==> Executing phase: 'bootstrap'
==> Executing phase: 'install'
==> Error: FileNotFoundError: [Errno 2] No such file or directory: 'output/bazel'

/lus/theta-fs0/projects/datascience/cadams/spack/cooley/spack/var/spack/repos/builtin/packages/bazel/package.py:136, in install:
        134    def install(self, spec, prefix):
        135        mkdir(prefix.bin)
  >>    136        install('output/bazel', prefix.bin)

See build log for details:
  /tmp/cadams/spack-stage/spack-stage-bazel-0.26.1-4mmpkcidyja4i6smxm2ahqifvh322xsz/spack-build-out.txt
```
Here was the command I have used:
```console
$ spack install py-tensorflow@1.15 cuda_arch=35 ^cuda@10.1 %gcc@6.3.0
```

And, I am on this system:
```
$ uname -a
Linux cc047 3.10.0-957.27.2.el7.x86_64 #1 SMP Tue Jul 9 16:53:14 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release
LSB Version:	:core-4.1-amd64:core-4.1-noarch
```

Any ideas for how to fix this?  I have the latest version of spack as of this morning.  Is there an older version of bazel that will work?

I have been trying again to install tensorflow.  Limited success, really.  I have tried with tensorflow versions 1.14.0 and 1.15.0, neither builds.

1.14.0 error:
```
==> Error: ProcessError: Command exited with status 1:
    'bazel' '--nohome_rc' '--nosystem_rc' '--output_user_root=/tmp/spack/tf' 'build' '--color=no' '--jobs=12' '--config=opt' '--verbose_failures' '--subcommands=pretty_print' '--explain=explainlogfile.txt' '--verbose_explanations' '--config=cuda' '--config=noaws' '--config=nogcp' '--config=nohdfs' '--config=noignite' '--config=nokafka' '--cxxopt=-D_GLIBCXX_USE_CXX11_ABI=0' '//tensorflow/tools/pip_package:build_pip_package'

5 errors found in build log:
     2195        -w \
     2196        '-std=c99' \
     2197        -c \
     2198        external/nasm/asm/rdstrnum.c \
     2199        -o \
     2200        bazel-out/k8-opt/bin/external/nasm/_objs/nasm/rdstrnum.o)
  >> 2201    ERROR: /tmp/spack/tf/966c9fe0240f93060470a6ff832dbefd/external/nasm/BUILD.bazel:8:1: C++ compilation of ru
             le '@@nasm//:nasm' failed (Exit 1): crosstool_wrapper_driver_is_not_gcc failed: error executing command
     2202      (cd /tmp/spack/tf/966c9fe0240f93060470a6ff832dbefd/execroot/org_tensorflow && \
     2203      exec env - \
     2204        CUDA_TOOLKIT_PATH=/gpfs/mira-fs0/software/x86_64/datascience/spack/opt/spack/linux-rhel7-haswell/gcc-8
             .2.0/cuda-10.1.243-d6wxkrjmwfuciuaqaaqesckso6cgjy2i \
     2205        CUDNN_INSTALL_PATH=/gpfs/mira-fs0/software/x86_64/datascience/spack/opt/spack/linux-rhel7-haswell/gcc-
             8.2.0/cudnn-7.6.5.32-10.1-linux-x64-uxogwqksn2qauwxcnnxjafzcktmt6h6h \
     2206        GCC_HOST_COMPILER_PATH=/soft/datascience/spack/lib/spack/env/gcc/gcc \
     2207        LD_LIBRARY_PATH=/gpfs/mira-fs0/software/x86_64/datascience/spack/opt/spack/linux-rhel7-haswell/gcc-8.2
             .0/cuda-10.1.243-d6wxkrjmwfuciuaqaaqesckso6cgjy2i/targets/x86_64-linux/lib/stubs:/gpfs/mira-fs0/software/x
             86_64/datascience/spack/opt/spack/linux-rhel7-haswell/gcc-8.2.0/cudnn-7.6.5.32-10.1-linux-x64-uxogwqksn2qa
             uwxcnnxjafzcktmt6h6h/lib64:/gpfs/mira-fs0/software/x86_64/datascience/spack/opt/spack/linux-rhel7-haswell/
             gcc-8.2.0/nccl-2.4.8-1-6o2d6dlye3qadynng6vetzcr4tm6seyx/lib \

     ...

     2248        TF_NEED_OPENCL_SYCL=0 \
     2249        TF_NEED_ROCM=0 \
     2250        TF_NEED_TENSORRT=0 \
     2251      external/local_config_cuda/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc -MD -MF bazel-out/k8-
             opt/bin/external/nasm/_objs/nasm/outelf.d '-frandom-seed=bazel-out/k8-opt/bin/external/nasm/_objs/nasm/out
             elf.o' -DHAVE_SNPRINTF -DHAVE_SYS_TYPES_H -iquote external/nasm -iquote bazel-out/k8-opt/bin/external/nasm
              -iquote external/bazel_tools -iquote bazel-out/k8-opt/bin/external/bazel_tools -isystem external/nasm/asm
              -isystem bazel-out/k8-opt/bin/external/nasm/asm -isystem external/nasm/include -isystem bazel-out/k8-opt/
             bin/external/nasm/include -isystem external/nasm/output -isystem bazel-out/k8-opt/bin/external/nasm/output
              -isystem external/nasm/x86 -isystem bazel-out/k8-opt/bin/external/nasm/x86 -Wno-builtin-macro-redefined '
             -D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -fPIE -U_FORTIFY_SOURCE '-D_FO
             RTIFY_SOURCE=1' -fstack-protector -Wall -fno-omit-frame-pointer -no-canonical-prefixes -fno-canonical-syst
             em-headers -DNDEBUG -g0 -O2 -ffunction-sections -fdata-sections '-march=haswell' '-mtune=haswell' -w '-std
             =c99' -c external/nasm/output/outelf.c -o bazel-out/k8-opt/bin/external/nasm/_objs/nasm/outelf.o)
     2252    Execution platform: @bazel_tools//platforms:host_platform
     2253    In file included from external/nasm/output/outelf.c:49:
  >> 2254    /gpfs/mira-fs0/software/x86_64/datascience/spack/opt/spack/linux-rhel7-haswell/gcc-8.2.0/python-3.7.4-j4sb
             fz26j6k2mt2vzmvxhtklskffam33/include/python3.7m/eval.h:10:12: error: unknown type name 'PyObject'
     2255     PyAPI_FUNC(PyObject *) PyEval_EvalCode(PyObject *, PyObject *, PyObject *);
     2256                ^~~~~~~~
  >> 2257    /gpfs/mira-fs0/software/x86_64/datascience/spack/opt/spack/linux-rhel7-haswell/gcc-8.2.0/python-3.7.4-j4sb
             fz26j6k2mt2vzmvxhtklskffam33/include/python3.7m/eval.h:12:12: error: unknown type name 'PyObject'
     2258     PyAPI_FUNC(PyObject *) PyEval_EvalCodeEx(PyObject *co,
     2259                ^~~~~~~~
  >> 2260    /gpfs/mira-fs0/software/x86_64/datascience/spack/opt/spack/linux-rhel7-haswell/gcc-8.2.0/python-3.7.4-j4sb
             fz26j6k2mt2vzmvxhtklskffam33/include/python3.7m/eval.h:21:12: error: unknown type name 'PyObject'
     2261     PyAPI_FUNC(PyObject *) _PyEval_EvalCodeWithName(
     2262                ^~~~~~~~
  >> 2263    /gpfs/mira-fs0/software/x86_64/datascience/spack/opt/spack/linux-rhel7-haswell/gcc-8.2.0/python-3.7.4-j4sb
             fz26j6k2mt2vzmvxhtklskffam33/include/python3.7m/eval.h:31:12: error: unknown type name 'PyObject'
     2264     PyAPI_FUNC(PyObject *) _PyEval_CallTracing(PyObject *func, PyObject *args);
     2265                ^~~~~~~~
     2266    Target //tensorflow/tools/pip_package:build_pip_package failed to build
     2267    INFO: Elapsed time: 57.328s, Critical Path: 1.32s
     2268    INFO: 8 processes: 8 local.
     2269    FAILED: Build did NOT complete successfully

See build log for details:
  /tmp/cadams/spack-stage/spack-stage-py-tensorflow-1.14.0-a3lc7hqjvpzmnv6zdhklnyvrczvldfty/spack-build-out.txt
```

1.15.0 error:
```
ERROR: /tmp/spack/tf/84085fcd4911be325444e10a40e1c6ce/external/com_google_protobuf/BUILD:106:1: undeclared inclusion(s) in rule '@com_google_protobuf//:protobuf_lite':
this rule is missing dependency declarations for the following files included by 'external/com_google_protobuf/src/google/protobuf/stubs/int128.cc':
  '/gpfs/mira-fs0/software/x86_64/datascience/spack/opt/spack/linux-rhel7-haswell/gcc-8.2.0/gettext-0.20.1-7ipvgryanyyat522m5ran5vsagy6bog2/include/libintl.h'
Target //tensorflow/tools/pip_package:build_pip_package failed to build
INFO: Elapsed time: 73.809s, Critical Path: 8.00s
INFO: 12 processes: 12 local.
FAILED: Build did NOT complete successfully
FAILED: Build did NOT complete successfully
```


This is on a standard GPU cluster, nothing complicated like cross compiling.  Bazel builds, at least, though as you mention it is not very reliable with the jdk download.

--- a/lib/spack/env/cc
+++ b/lib/spack/env/cc
@@ -273,12 +273,24 @@ system_libdirs=()
 system_rpaths=()
 libs=()
 other_args=()
+isystem_system_includes=()
+isystem_includes=()
 
 while [ -n "$1" ]; do
     # an RPATH to be added after the case statement.
     rp=""
 
     case "$1" in
+        -isystem*)
+            arg="${1#-isystem}"
+	    isystem_was_used=true
+            if [ -z "$arg" ]; then shift; arg="$1"; fi
+            if system_dir "$arg"; then
+                isystem_system_includes+=("$arg")
+            else
+                isystem_includes+=("$arg")
+            fi
+            ;;
         -I*)
             arg="${1#-I}"
             if [ -z "$arg" ]; then shift; arg="$1"; fi
@@ -425,12 +437,6 @@ then
     esac
 fi
 
-# Prepend include directories
-IFS=':' read -ra include_dirs <<< "$SPACK_INCLUDE_DIRS"
-if [[ $mode == cpp || $mode == cc || $mode == as || $mode == ccld ]]; then
-    includes=("${includes[@]}" "${include_dirs[@]}")
-fi
-
 IFS=':' read -ra rpath_dirs <<< "$SPACK_RPATH_DIRS"
 if [[ $mode == ccld || $mode == ld ]]; then
 
@@ -481,9 +487,22 @@ args=()
 # flags assembled earlier
 args+=("${flags[@]}")
 
-# include directory search paths
+# Insert include directories just prior to any system include directories
+IFS=':' read -ra spack_include_dirs <<< "$SPACK_INCLUDE_DIRS"
+
 for dir in "${includes[@]}";         do args+=("-I$dir"); done
+for dir in "${isystem_includes[@]}";         do args+=("-isystem$dir"); done
+
+if [[ $mode == cpp || $mode == cc || $mode == as || $mode == ccld ]]; then
+    if [[ "$isystem_was_used" == "true" ]] ; then
+	for dir in "${spack_include_dirs[@]}";  do args+=("-isystem$dir"); done
+    else
+	for dir in "${spack_include_dirs[@]}";  do args+=("-I$dir"); done
+    fi
+fi
+
 for dir in "${system_includes[@]}";  do args+=("-I$dir"); done
+for dir in "${isystem_system_includes[@]}";  do args+=("-isystem$dir"); done
 
 # Library search paths
 for dir in "${libdirs[@]}";          do args+=("-L$dir"); done

--- a/lib/spack/env/cc
+++ b/lib/spack/env/cc
@@ -15,9 +15,9 @@
 # 1. It allows Spack to swap compilers into and out of builds easily.
 # 2. It adds several options to the compile line so that spack
 #    packages can find their dependencies at build time and run time:
-#      -I           arguments for dependency /include directories.
-#      -L           arguments for dependency /lib directories.
-#      -Wl,-rpath   arguments for dependency /lib directories.
+#      -I and/or -isystem arguments for dependency /include directories.
+#      -L                 arguments for dependency /lib directories.
+#      -Wl,-rpath         arguments for dependency /lib directories.
 #
 
 # This is an array of environment variables that need to be set before
@@ -251,10 +251,11 @@ input_command="$*"
 #
 # Parse the command line arguments.
 #
-# We extract -L, -I, and -Wl,-rpath arguments from the command line and
-# recombine them with Spack arguments later.  We parse these out so that
-# we can make sure that system paths come last, that package arguments
-# come first, and that Spack arguments are injected properly.
+# We extract -L, -I, -isystem and -Wl,-rpath arguments from the
+# command line and recombine them with Spack arguments later.  We
+# parse these out so that we can make sure that system paths come
+# last, that package arguments come first, and that Spack arguments
+# are injected properly.
 #
 # All other arguments, including -l arguments, are treated as
 # 'other_args' and left in their original order.  This ensures that
@@ -488,11 +489,11 @@ args=()
 args+=("${flags[@]}")
 
 # Insert include directories just prior to any system include directories
-IFS=':' read -ra spack_include_dirs <<< "$SPACK_INCLUDE_DIRS"
 
 for dir in "${includes[@]}";         do args+=("-I$dir"); done
 for dir in "${isystem_includes[@]}";         do args+=("-isystem$dir"); done
 
+IFS=':' read -ra spack_include_dirs <<< "$SPACK_INCLUDE_DIRS"
 if [[ $mode == cpp || $mode == cc || $mode == as || $mode == ccld ]]; then
     if [[ "$isystem_was_used" == "true" ]] ; then
 	for dir in "${spack_include_dirs[@]}";  do args+=("-isystem$dir"); done

--- a/lib/spack/spack/test/cc.py
+++ b/lib/spack/spack/test/cc.py
@@ -338,6 +338,33 @@ def test_ccld_deps():
             test_args_without_paths)
 
 
+def test_ccld_deps_isystem():
+    """Ensure all flags are added in ccld mode,
+       when a build uses -isystem."""
+    with set_env(SPACK_INCLUDE_DIRS='xinc:yinc:zinc',
+                 SPACK_RPATH_DIRS='xlib:ylib:zlib',
+                 SPACK_LINK_DIRS='xlib:ylib:zlib'):
+        mytest_args = test_args + ['-isystemfooinc']
+        check_args(
+            cc, mytest_args,
+            [real_cc] +
+            test_include_paths +
+            ['-isystemfooinc',
+             '-isystemxinc',
+             '-isystemyinc',
+             '-isystemzinc'] +
+            test_library_paths +
+            ['-Lxlib',
+             '-Lylib',
+             '-Lzlib'] +
+            ['-Wl,--disable-new-dtags'] +
+            test_wl_rpaths +
+            ['-Wl,-rpath,xlib',
+             '-Wl,-rpath,ylib',
+             '-Wl,-rpath,zlib'] +
+            test_args_without_paths)
+
+
 def test_cc_deps():
     """Ensure -L and RPATHs are not added in cc mode."""
     with set_env(SPACK_INCLUDE_DIRS='xinc:yinc:zinc',
@@ -350,6 +377,26 @@ def test_cc_deps():
             ['-Ixinc',
              '-Iyinc',
              '-Izinc'] +
+            test_library_paths +
+            ['-c'] +
+            test_args_without_paths)
+
+
+def test_cc_deps_isystem():
+    """Ensure -L and RPATHs are not added in cc mode,
+       when a build uses -isystem"""
+    with set_env(SPACK_INCLUDE_DIRS='xinc:yinc:zinc',
+                 SPACK_RPATH_DIRS='xlib:ylib:zlib',
+                 SPACK_LINK_DIRS='xlib:ylib:zlib'):
+        mytest_args = test_args + ['-isystemfooinc']
+        check_args(
+            cc, ['-c'] + mytest_args,
+            [real_cc] +
+            test_include_paths +
+            ['-isystemfooinc',
+             '-isystemxinc',
+             '-isystemyinc',
+             '-isystemzinc'] +
             test_library_paths +
             ['-c'] +
             test_args_without_paths)
@@ -375,6 +422,42 @@ def test_ccld_with_system_dirs():
              '-Izinc'] +
             ['-I/usr/include',
              '-I/usr/local/include'] +
+            test_library_paths +
+            ['-Lxlib',
+             '-Lylib',
+             '-Lzlib'] +
+            ['-L/usr/local/lib',
+             '-L/lib64/'] +
+            ['-Wl,--disable-new-dtags'] +
+            test_wl_rpaths +
+            ['-Wl,-rpath,xlib',
+             '-Wl,-rpath,ylib',
+             '-Wl,-rpath,zlib'] +
+            ['-Wl,-rpath,/usr/lib64'] +
+            test_args_without_paths)
+
+
+def test_ccld_with_system_dirs_isystem():
+    """Ensure all flags are added in ccld mode,
+       when a build uses -isystem"""
+    with set_env(SPACK_INCLUDE_DIRS='xinc:yinc:zinc',
+                 SPACK_RPATH_DIRS='xlib:ylib:zlib',
+                 SPACK_LINK_DIRS='xlib:ylib:zlib'):
+
+        sys_path_args = ['-isystem/usr/include',
+                         '-L/usr/local/lib',
+                         '-Wl,-rpath,/usr/lib64',
+                         '-isystem/usr/local/include',
+                         '-L/lib64/']
+        check_args(
+            cc, sys_path_args + test_args,
+            [real_cc] +
+            test_include_paths +
+            ['-isystemxinc',
+             '-isystemyinc',
+             '-isystemzinc'] +
+            ['-isystem/usr/include',
+             '-isystem/usr/local/include'] +
             test_library_paths +
             ['-Lxlib',
              '-Lylib',

--- a/lib/spack/spack/test/cc.py
+++ b/lib/spack/spack/test/cc.py
@@ -339,8 +339,11 @@ def test_ccld_deps():
 
 
 def test_ccld_deps_isystem():
-    """Ensure all flags are added in ccld mode,
-       when a build uses -isystem."""
+    """Ensure all flags are added in ccld mode.
+       When a build uses -isystem, Spack should inject it's
+       include paths using -isystem. Spack will insert these
+       after any provided -isystem includes, but before any
+       system directories included using -isystem"""
     with set_env(SPACK_INCLUDE_DIRS='xinc:yinc:zinc',
                  SPACK_RPATH_DIRS='xlib:ylib:zlib',
                  SPACK_LINK_DIRS='xlib:ylib:zlib'):
@@ -382,26 +385,6 @@ def test_cc_deps():
             test_args_without_paths)
 
 
-def test_cc_deps_isystem():
-    """Ensure -L and RPATHs are not added in cc mode,
-       when a build uses -isystem"""
-    with set_env(SPACK_INCLUDE_DIRS='xinc:yinc:zinc',
-                 SPACK_RPATH_DIRS='xlib:ylib:zlib',
-                 SPACK_LINK_DIRS='xlib:ylib:zlib'):
-        mytest_args = test_args + ['-isystemfooinc']
-        check_args(
-            cc, ['-c'] + mytest_args,
-            [real_cc] +
-            test_include_paths +
-            ['-isystemfooinc',
-             '-isystemxinc',
-             '-isystemyinc',
-             '-isystemzinc'] +
-            test_library_paths +
-            ['-c'] +
-            test_args_without_paths)
-
-
 def test_ccld_with_system_dirs():
     """Ensure all flags are added in ccld mode."""
     with set_env(SPACK_INCLUDE_DIRS='xinc:yinc:zinc',
@@ -438,8 +421,10 @@ def test_ccld_with_system_dirs():
 
 
 def test_ccld_with_system_dirs_isystem():
-    """Ensure all flags are added in ccld mode,
-       when a build uses -isystem"""
+    """Ensure all flags are added in ccld mode.
+       Ensure that includes are in the proper
+       place when a build uses -isystem, and uses
+       system directories in the include paths"""
     with set_env(SPACK_INCLUDE_DIRS='xinc:yinc:zinc',
                  SPACK_RPATH_DIRS='xlib:ylib:zlib',
                  SPACK_LINK_DIRS='xlib:ylib:zlib'):

